### ISSUE TITLE
Visual Studio integration is unclear and outdated

### ISSUE THREAD

**Issue by Trider12** [godotengine/godot-docs#6911]
**Godot version: 4.0**

**[URL to the documentation page](https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_basics.html#configuring-an-external-editor)**

**Issue description:**
1. Now that Godot targets .Net 6.0, VS 2019 is not supported and should be removed from the available editors list.
2. The docs say to install `Mobile development with .NET` and `.NET Core cross-platform development`, but both of them are now out of support and, I believe, are not required due to the transition to .Net 6.0 (I may be wrong here).
![image](https://user-images.githubusercontent.com/47952308/223108483-fc4f93fd-41d0-4249-9eaf-9b3a7f90cf9d.png)

4. The docs mention the VS [extension](https://github.com/godotengine/godot-csharp-visualstudio) which (has been barely working for 3.x) is [obsolete](https://github.com/godotengine/godot-csharp-visualstudio/issues/42#issuecomment-1250411467) for 4.x.
5. There is no information on how to actually debug the game with VS. There's this [hack](https://www.reddit.com/r/GodotCSharp/comments/xgpqfh/oc_rundebug_godot4_c_projects_from_visual_studio/), but it is a hack, has its issues and doesn't look like a proper solution. If it is the proper solution, then it needs to be documented.

**Comment by silenuz** [comment]
In Visual Studio 2022 it's pretty easy.  Right click the project, choose properties.

Under properties select Debug, and then click the link to open the Debug launch profiles dialog.

Delete the current profile, and add an executable profile.  Set the executable path to point to the main Godot exe.

Set arguments to be --path . (or --path "/absolute/path/to/project/folder")  Note the quotes.

Enable native debugging and thats it.  Should be able to now launch a debug session with Visual Studio.

If you want to also be able to debug editor tools add another profile and add --editor to the argument line.

![0-Visual-studio-debug](https://user-images.githubusercontent.com/13521146/223510644-d395f966-90aa-4361-ad36-4c8cdef9ca17.gif)

**Comment by Trider12** [comment]
@silenuz Yeah, I did exactly that (even in VS2019 using this [hack](https://learn.microsoft.com/en-us/answers/questions/620284/how-to-target-net-6-0-with-visual-studio-2019)), but my point is that this needs to be in the documentation, not on Reddit/GitHub.

**Comment by silenuz** [comment]
Maybe someone can integrate the attached.  I'm in the middle of trying to do some CSharp tutorials for Godot and it's my preface.  
[debug-csharp.zip](https://github.com/godotengine/godot-docs/files/10913024/debug-csharp.zip)

**Comment by GeorgeS2019** [comment]
@Trider12 

==> [March 2023 Update ](https://github.com/godotengine/godot-csharp-visualstudio/issues/48)to Godot4 debugging using VS2022

**Comment by silenuz** [comment]
If anyone cares I ended up creating a DotNet project template for Godot.  Usage is simple:
`dotnet new Godot -o {NameOfProject} -G {PathToGodtotExe}`

This will create a project that already has the necessary launch settings for running or debugging your code in Visual Studio or Jetbrains' Rider.  Note once installed the template will also be available in the above IDEs.

After the project is generated simply open it in the Godot editor as usual.

I'm currently working on packaging for easier use, and hope to have that finished by the end of the weekend.
[Godot project Template](https://github.com/silenuz/GodotNetProjectTemplate)

**Comment by skyace65** [comment]
All issues mentioned here have been fixed as of the current stable version of this page.
